src/components/Button/Button.tsx

```diff
@@ -1,37 +1,65 @@
+import "./Button.scss";
 import { ReactNode } from "react";
 import cx from "classnames";
-import "./Button.scss";
+import ButtonLink from "./ButtonLink";
+
+type ButtonVariant = "primary" | "primary-action" | "semi-clear" | "clear";
 
-type Props = {
+type ButtonProps = {
   children: ReactNode;
-  imgSrc?: string;
-  imgName?: string;
+  variant: ButtonVariant;
   className?: string;
-  href?: string;
-  onClick?: () => void;
   disabled?: boolean;
+  onClick?: () => void;
+  to?: string;
+  type?: "button" | "submit" | "reset";
+  imgInfo?: {
+    src: string;
+    alt?: string;
+  };
 };
 
-export default function Button({ href, imgSrc, children, className, imgName, onClick, disabled = false }: Props) {
-  let classNames = cx("btn", className);
+export default function Button({
+  variant,
+  disabled,
+  onClick,
+  children,
+  to,
+  className,
+  imgInfo,
+  type,
+  ...rest
```

Comment:
looks like ...rest will contain nothing, since all the props are distracted?
maybe jsut remove it?
If you wanted to pass all the props for the html button
then the type should be

```
type ButtonProps =  React.HTMLProps<HTMLButtonElement> & {...}
```

Reply:
Thanks, this is a much better way 🙇‍♂️